src/EnergyPlus/ZoneEquipmentManager.cc

```diff
@@ -3508,74 +3519,341 @@ namespace ZoneEquipmentManager {
+							} else {
+								moisture.SequencedOutputRequired( equipNum ) = moisture.TotalOutputRequired  * plr;
+								moisture.SequencedOutputRequiredToHumidSP( equipNum ) = moisture.OutputRequiredToHumidifyingSP * plr;
+							}
+							moisture.SequencedOutputRequiredToDehumidSP( equipNum ) = 0.0;
+						} else {
+							energy.SequencedOutputRequired( equipNum ) = 0.0;
+							energy.SequencedOutputRequiredToHeatingSP( equipNum ) = 0.0;
+							energy.SequencedOutputRequiredToCoolingSP( equipNum ) = 0.0;
+							moisture.SequencedOutputRequired( equipNum ) = 0.0;
+							moisture.SequencedOutputRequiredToHumidSP( equipNum ) = 0.0;
+							moisture.SequencedOutputRequiredToDehumidSP( equipNum ) = 0.0;
+						}
+					} else {
+						if ( thisZEqList.CoolingPriority( equipNum ) > 0 ) {
+							energy.SequencedOutputRequired( equipNum ) = thisZEqList.CoolingCapacity( equipNum ) * plr;
+							energy.SequencedOutputRequiredToHeatingSP( equipNum ) = thisZEqList.CoolingCapacity( equipNum ) * plr;
+							energy.SequencedOutputRequiredToCoolingSP( equipNum ) = thisZEqList.CoolingCapacity( equipNum ) * plr;
+							if ( energy.OutputRequiredToCoolingSP != 0.0 ) {
+								moisture.SequencedOutputRequired( equipNum ) = moisture.TotalOutputRequired * ( thisZEqList.CoolingCapacity( equipNum ) * plr ) / energy.OutputRequiredToCoolingSP;
+								moisture.SequencedOutputRequiredToDehumidSP( equipNum ) = moisture.OutputRequiredToDehumidifyingSP * ( thisZEqList.CoolingCapacity( equipNum ) * plr ) / energy.OutputRequiredToCoolingSP;
+							} else {
+								moisture.SequencedOutputRequired( equipNum ) = moisture.TotalOutputRequired  * plr;
+								moisture.SequencedOutputRequiredToDehumidSP( equipNum ) = moisture.OutputRequiredToDehumidifyingSP * plr;
+							}
+							moisture.SequencedOutputRequiredToHumidSP( equipNum ) = 0.0;
+						} else {
+							energy.SequencedOutputRequired( equipNum ) = 0.0;
+							energy.SequencedOutputRequiredToHeatingSP( equipNum ) = 0.0;
+							energy.SequencedOutputRequiredToCoolingSP( equipNum ) = 0.0;
+							moisture.SequencedOutputRequired( equipNum ) = 0.0;
+							moisture.SequencedOutputRequiredToHumidSP( equipNum ) = 0.0;
+							moisture.SequencedOutputRequiredToDehumidSP( equipNum ) = 0.0;
+						}
+					}
+				}
+				// Set loads to zero for remaining equipment
+				for ( int equipNum = numOperating + 1; equipNum <= thisZEqList.NumOfEquipTypes; ++equipNum ) {
+					energy.SequencedOutputRequired( equipNum ) = 0.0;
+					energy.SequencedOutputRequiredToHeatingSP( equipNum ) = 0.0;
+					energy.SequencedOutputRequiredToCoolingSP( equipNum ) = 0.0;
+					moisture.SequencedOutputRequired( equipNum ) = 0.0;
+					moisture.SequencedOutputRequiredToHumidSP( equipNum ) = 0.0;
+					moisture.SequencedOutputRequiredToDehumidSP( equipNum ) = 0.0;
+				}
+				break;
+			default:
+				ShowFatalError( "DistributeSystemOutputRequired: Illegal load distribution scheme type." );
+				break;
+		}
```

Comment:
I wonder if there's some future possibility of combining the plant algorithm and this one together.  Would be great...